Expensify/App
# Add better error handling for Onfido

cc @Jag96 @ctkochan22 I couldn't get Onfido to work on desktop or web no matter how hard I tried to take a picture of my ID via my phone. But tested this with the current web-secure changes and it caught the weird error about referrer.

### Details

Onfido was throwing a pretty clear error but we are not logging it at all or handling it to allow the user to attempt the flow again. This will log these errors and also try to move the user back to the Company Step so they can attempt to go through Onfido again.

### Fixed Issues (Only related to does not fix)
https://github.com/Expensify/Expensify/issues/177787#

### Tests (iOS only)

In order to test this we need to force the Web-Secure API to do something it shouldn't do like set a bad application id or referrer. For this test I just applied this diff in Web-Secure before testing:

```diff
diff --git a/lib/Onfido/API.php b/lib/Onfido/API.php
index 957f53ac..223dad75 100644
--- a/lib/Onfido/API.php
+++ b/lib/Onfido/API.php
@@ -195,6 +195,10 @@ class Onfido_API
         $api = self::getOnfidoApi();
         $sdk_token_request = new Onfido_SdkTokenRequest();
         $sdk_token_request->setApplicantId($applicantID);
+
+        // Add incorrect application id
+        $sdk_token_request->setApplicationId('pizza');
+
         $response = $api->generateSdkToken($sdk_token_request);
         if ($response instanceof Onfido_SdkTokenResponse) {
             return $response->getToken() ?? '';
```

1. Create a new account with a non-public domain on the `freePlan` beta
2. Create a workspace
3. Tap "Expensify Card"
3. Press "Get Started"
4. Select the "Add account manually" option
5. Follow instructions for creating a Verifying bank account [here](https://stackoverflow.com/c/expensify/questions/342)
2. When you reach the Onfido step cancel out of the flow by pressing the blue arrow to the left
3. Verify you are brought back to the company step
4. Enter the password to move to the next step again
5. Press start button and allow camera access
6. Take a picture of anything
7. The upload should fail and you should be redirected back to the company step + an error growl should appear like so:
![2021-09-20_11-54-17](https://user-images.githubusercontent.com/32969087/134081030-190445f7-14ae-4687-aed6-2057b32ac598.png)
8. Put the app into the background so logs are flushed and tail to find the log that says:
>Unexpected application_id: com.chat.expensify.chat, please check the application_id used when generating the token

#### Other Platforms

Test we can upload documents without issue with and without the diff applied.

### QA Steps
None

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
